cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -444,6 +444,25 @@ public void deleteApp(HttpRequest request, HttpResponder responder,
     responder.sendStatus(HttpResponseStatus.OK);
   }
 
+  /**
+   * Deletes all application state stored in CDAP state store for the given application.
+   *
+   * @param request The {@link HttpRequest}.
+   * @param responder The {@link HttpResponder}.
+   * @param namespaceId Namespace id string.
+   * @param appName App name string.
+   * @throws Exception Any {@link Exception} encountered.
+   */
+  @DELETE
+  @Path("/apps/{app-id}/state")
+  public void deleteAppState(HttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespaceId,
+      @PathParam("app-id") final String appName) throws Exception {
+    LOG.debug("Deleting all application state for {} in namespace {}", appName, namespaceId);
+    applicationLifecycleService.deleteAllStates(new NamespaceId(namespaceId), appName);
```

Comment:
since this isn't an internal call anymore, we need to use the accessEnforcer in the deleteAllStates() method to make sure the user is allowed to delete. May be good to check with @dli357 about this

Reply:
Added check for Application UPDATE permission for state deletion.